monai/transforms/transforms.py

```diff
@@ -441,10 +444,514 @@ def __call__(self, img):
+                 as_tensor_output=True,
+                 device=None):
+        """
+        Args:
+            rotate_range (a sequence of positive floats): rotate_range[0] with be used to generate the 1st rotation
+                parameter from `uniform[-rotate_range[0], rotate_range[0])`. Similarly, `rotate_range[2]` and
+                `rotate_range[3]` are used in 3D affine for the range of 2nd and 3rd axes.
+            shear_range (a sequence of positive floats): shear_range[0] with be used to generate the 1st shearing
+                parameter from `uniform[-shear_range[0], shear_range[0])`. Similarly, `shear_range[1]` to
+                `shear_range[N]` controls the range of the uniform distribution used to generate the 2nd to
+                N-th parameter.
+            translate_range (a sequence of positive floats): translate_range[0] with be used to generate the 1st
+                shift parameter from `uniform[-translate_range[0], translate_range[0])`. Similarly, `translate_range[1]`
+                to `translate_range[N]` controls the range of the uniform distribution used to generate
+                the 2nd to N-th parameter.
+            scale_range (a sequence of positive floats): scaling_range[0] with be used to generate the 1st scaling
+                factor from `uniform[-scale_range[0], scale_range[0]) + 1.0`. Similarly, `scale_range[1]` to
+                `scale_range[N]` controls the range of the uniform distribution used to generate the 2nd to
+                N-th parameter.
+
+        See also:
+            `from monai.transforms.utils import (create_rotate, create_shear, create_translate, create_scale)`
+        """
+        self.rotate_range = ensure_tuple(rotate_range)
+        self.shear_range = ensure_tuple(shear_range)
+        self.translate_range = ensure_tuple(translate_range)
+        self.scale_range = ensure_tuple(scale_range)
+
+        self.rotate_params = None
+        self.shear_params = None
+        self.translate_params = None
+        self.scale_params = None
+
+        self.as_tensor_output = as_tensor_output
+        self.device = device
+
+    def randomize(self):
+        if self.rotate_range:
+            self.rotate_params = [self.R.uniform(-f, f) for f in self.rotate_range if f is not None]
+        if self.shear_range:
+            self.shear_params = [self.R.uniform(-f, f) for f in self.shear_range if f is not None]
+        if self.translate_range:
+            self.translate_params = [self.R.uniform(-f, f) for f in self.translate_range if f is not None]
+        if self.scale_range:
+            self.scale_params = [self.R.uniform(-f, f) + 1.0 for f in self.scale_range if f is not None]
+
+    def __call__(self, spatial_size=None, grid=None):
+        """
+        Returns:
+            a 2D (3xHxW) or 3D (4xHxWxD) grid.
+        """
+        self.randomize()
+        _affine_grid = AffineGrid(self.rotate_params, self.shear_params, self.translate_params, self.scale_params,
```

Comment:
I think we should not use "_" for local variables. It's used for protected or private member variables.